packages/components/src/components/form/Textarea/Textarea.tsx

```diff
@@ -68,12 +79,26 @@ export const Textarea = ({
     isMonospace,
     noTopLabel,
     labelRight,
+    characterCount,
     noError,
     children,
     ...rest
 }: TextareaProps) => {
     const [isHovered, setIsHovered] = useState(false);
 
+    const getCharacterCount = () => {
+        // controlled component
+        if (characterCount === true && value !== undefined && maxLength !== undefined) {
+            return `${value.length} / ${maxLength}`;
+        }
+        // uncontrolled component
+        if (typeof characterCount === 'object') {
+            return `${characterCount.current} / ${characterCount.max}`;
+        }
+    };
+
+    const formattedCharacterCount = getCharacterCount();
+
```

Comment:
Much more cumbersome but allows for a more frictionless component API, since you don't have to care, just pass a prop :)

```suggestion
     const [charCount, setCharCount] = useState();

    const textareRef = useRef<unknown>(null);

    useEffect(() => {
        if (characterCount && charCount === undefined) {
            setCharCount(textareRef.current?.value.length);
        }
    }, [characterCount, charCount, textareRef]);

    useEffect(() => {
        if (characterCount) {
            const handleInput = (e: Event) => setCharCount(e.target?.value?.length);

            (textareRef.current as HTMLTextAreaElement).addEventListener('input', handleInput);

            return () =>
                (textareRef.current as HTMLTextAreaElement).removeEventListener(
                    'input',
                    handleInput,
                );
        }
    }, [characterCount]);

```

Reply:
The suggested code is incomplete but even after adjustments it does not work in every situation, see video:

https://github.com/trezor/trezor-suite/assets/42465546/631fecb3-96e3-42fc-88b0-b5aa0d5478eb

I would like to improve the API, but couldn't figure out how because refs should't be used as `useEffect` dependencies and `onChange` is not always triggered.

Reply:
Apoligies for the incomplete code! And yes, I couldn't make it work in this for either :\ It is sadly not reacting to the value simply being assigned...